tox.ini

```diff
@@ -0,0 +1,14 @@
+[tox]
+envlist = py39, py310, py311, py312
+labels =
+    ; Used to map GitHub workflow python version to tox env
+    3.9 = py39
+    3.10 = py310
+    3.11 = py311
+    3.12 = py312
+
+[testenv]
+deps =
+    pytest>=8
+commands =
+    pytest tests/ --import-mode=importlib
```

Comment:
`--import-mode=importlib` tells pytest to use the `importlib` machinery when importing test modules. Pretty much it's what we do with `tox` since it's not guaranteed to run within a `poetry` environment (like in CI). 

We don't need to use it when developing locally - we still run `poetry run pytest` or just `pytest` if we're within a poetry virtual environment :)